packages/language-service/lib/index.js

```diff
@@ -0,0 +1,891 @@
+      syncSnapshot(fileName)
+      const diagnostics = ls.getSuggestionDiagnostics(fileName)
+
+      for (const diagnostic of diagnostics) {
+        patchDiagnostic(diagnostic)
+      }
+
+      return diagnostics
+    },
+
+    getSyntacticClassifications(fileName, span) {
+      assertNotMdx(fileName, 'getSyntacticClassifications')
+      return ls.getSyntacticClassifications(fileName, span)
+    },
+
+    getSyntacticDiagnostics(fileName) {
+      const snapshot = syncSnapshot(fileName)
+      if (snapshot?.error) {
+        return toDiagnostic(ts, snapshot.error)
+      }
+
+      const diagnostics = ls.getSyntacticDiagnostics(fileName)
+
+      patchDiagnosticsWithLocation(diagnostics)
+
+      return diagnostics
+    },
+
+    getTodoComments(fileName, descriptors) {
+      assertNotMdx(fileName, 'getTodoComments')
+      return ls.getTodoComments(fileName, descriptors)
+    },
+
+    getTypeDefinitionAtPosition(fileName, position) {
+      const snapshot = syncSnapshot(fileName)
+
+      const definition = ls.getDefinitionAtPosition(
+        fileName,
+        snapshot?.getShadowPosition(position) ?? position
+      )
+
+      if (definition) {
+        patchDocumentSpans(definition)
+      }
+
+      return definition
+    },
+
+    isValidBraceCompletionAtPosition(fileName, position, openingBrace) {
+      assertNotMdx(fileName, 'isValidBraceCompletionAtPosition')
+      return ls.isValidBraceCompletionAtPosition(
+        fileName,
+        position,
+        openingBrace
+      )
+    },
+
+    organizeImports(args, formatOptions, preferences) {
+      assertNotMdx(args.fileName, 'organizeImports')
+      return ls.organizeImports(args, formatOptions, preferences)
+    },
+
+    prepareCallHierarchy(fileName, position) {
+      assertNotMdx(fileName, 'prepareCallHierarchy')
+      return ls.prepareCallHierarchy(fileName, position)
+    },
+
+    provideCallHierarchyIncomingCalls(fileName, position) {
+      assertNotMdx(fileName, 'provideCallHierarchyIncomingCalls')
+      return ls.provideCallHierarchyIncomingCalls(fileName, position)
+    },
+
+    provideCallHierarchyOutgoingCalls(fileName, position) {
+      assertNotMdx(fileName, 'provideCallHierarchyOutgoingCalls')
+      return ls.provideCallHierarchyOutgoingCalls(fileName, position)
+    },
+
+    provideInlayHints(fileName, span, preferences) {
+      assertNotMdx(fileName, 'provideInlayHints')
+      return ls.provideInlayHints(fileName, span, preferences)
+    },
+
+    toggleLineComment(fileName, textRange) {
+      assertNotMdx(fileName, 'toggleLineComment')
+      return ls.toggleLineComment(fileName, textRange)
+    },
+
+    toggleMultilineComment(fileName, textRange) {
+      assertNotMdx(fileName, 'toggleMultilineComment')
+      return ls.toggleMultilineComment(fileName, textRange)
+    },
+
+    uncommentSelection(fileName, textRange) {
+      assertNotMdx(fileName, 'uncommentSelection')
+      return ls.uncommentSelection(fileName, textRange)
+    }
+  }
+}
```

Comment:
This is rather long, so perhaps split it up, but it also looks like the code could be generated with code?

Reply:
Do you mean something like this?

```js
{
  // …
  toggleLineComment: notImplemented('toggleLineComment'),
  toggleMultilineComment: notImplemented('toggleMultilineComment'),
  uncommentSelection: notImplemented('uncommentSelection')
}
```

We could also loop over the keys of `ls`, but I kind of like mentioning each potential method explicitly. It shows what other potential functionality could be implemented later.

Reply:
Sure, factories for the ones that are not implemented, but also factories for things that just pass arguments through!
Listing everything is fine by me

Reply:
The only one that _just passes arguments through_ is `applyCodeActionCommand`, and that’s only because we can’t determine the fileName passed there. Otherwise it would be marked the same way.